frame/support/src/traits/dispatch.rs

```diff
@@ -250,6 +318,24 @@ impl<OuterOrigin, L: EnsureOrigin<OuterOrigin>, R: EnsureOrigin<OuterOrigin>>
 		L::try_successful_origin().or_else(|()| R::try_successful_origin())
 	}
 }
+impl<
+		OuterOrigin,
+		L: EnsureOriginWithArg<OuterOrigin, Argument>,
+		R: EnsureOriginWithArg<OuterOrigin, Argument, Success = L::Success>,
```

Comment:
```suggestion
		R: EnsureOriginWithArg<OuterOrigin, Argument>,
```
Otherwise it is not diverse.